// @azure/msal-angular/esm2015/msal.redirect.component.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License.
 */
/**
 * This is a dedicated redirect component to be added to Angular apps to
 * handle redirects when using @azure/msal-angular.
 * Import this component to use redirects in your app.
 */
import { Component } from "@angular/core";
import { MsalService } from "./msal.service";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './msal.service';
export class MsalRedirectComponent {
    constructor(authService) {
        this.authService = authService;
    }
    ngOnInit() {
        this.authService.getLogger().verbose("MsalRedirectComponent activated");
        this.authService.handleRedirectObservable().subscribe();
    }
}
MsalRedirectComponent.ɵfac = function MsalRedirectComponent_Factory(t) { return new (t || MsalRedirectComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.MsalService)); };
MsalRedirectComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MsalRedirectComponent, selectors: [["app-redirect"]], decls: 0, vars: 0, template: function MsalRedirectComponent_Template(rf, ctx) { }, encapsulation: 2 });
MsalRedirectComponent.ctorParameters = () => [
    { type: MsalService }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(MsalRedirectComponent, [{
        type: Component,
        args: [{
                selector: "app-redirect",
                template: ""
            }]
    }], function () { return [{ type: ɵngcc1.MsalService }]; }, null); })();
