// @progress/kendo-angular-navigation/dist/npm/appbar.module.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var appbar_component_1 = require("./appbar/appbar.component");
var appbar_section_component_1 = require("./appbar/appbar-section.component");
var appbar_spacer_component_1 = require("./appbar/appbar-spacer.component");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./appbar/appbar.component');
var ɵngcc2 = require('./appbar/appbar-section.component');
var ɵngcc3 = require('./appbar/appbar-spacer.component');
var ɵngcc4 = require('@angular/common');
var exportedModules = [
    appbar_component_1.AppBarComponent,
    appbar_section_component_1.AppBarSectionComponent,
    appbar_spacer_component_1.AppBarSpacerComponent
];
var declarations = exportedModules.slice();
/**
 * Represents the [NgModule]({{ site.data.urls.angular['ngmoduleapi'] }})
 * definition for the AppBar component.
 *
 * * @example
 *
 * ```ts-no-run
 * // Import the AppBar module
 * import { AppBarModule } from '@progress/kendo-angular-navigation';
 *
 * // The browser platform with a compiler
 * import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 * import { BrowserModule } from '@angular/platform-browser';
 *
 * import { NgModule } from '@angular/core';
 *
 * // Import the app component
 * import { AppComponent } from './app.component';
 *
 * // Define the app module
 * _@NgModule({
 *     declarations: [AppComponent], // declare app component
 *     imports:      [BrowserModule, AppBarModule], // import AppBar module
 *     bootstrap:    [AppComponent]
 * })
 * export class AppModule {}
 *
 * // Compile and launch the module
 * platformBrowserDynamic().bootstrapModule(AppModule);
 *
 * ```
 */
var AppBarModule = /** @class */ (function () {
    function AppBarModule() {
    }
AppBarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AppBarModule });
AppBarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AppBarModule_Factory(t) { return new (t || AppBarModule)(); }, imports: [[common_1.CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AppBarModule, { declarations: [ɵngcc1.AppBarComponent, ɵngcc2.AppBarSectionComponent, ɵngcc3.AppBarSpacerComponent], imports: [ɵngcc4.CommonModule], exports: [ɵngcc1.AppBarComponent, ɵngcc2.AppBarSectionComponent, ɵngcc3.AppBarSpacerComponent] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AppBarModule, [{
        type: core_1.NgModule,
        args: [{
                declarations: [declarations],
                exports: [exportedModules],
                imports: [common_1.CommonModule]
            }]
    }], function () { return []; }, null); })();
    return AppBarModule;
}());
exports.AppBarModule = AppBarModule;
