// @progress/kendo-angular-navigation/dist/npm/bottomnavigation.module.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var bottomnavigation_item_component_1 = require("./bottomnavigation/bottomnavigation-item.component");
var item_template_directive_1 = require("./bottomnavigation/templates/item-template.directive");
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var bottomnavigation_component_1 = require("./bottomnavigation/bottomnavigation.component");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./bottomnavigation/bottomnavigation-item.component');
var ɵngcc2 = require('./bottomnavigation/bottomnavigation.component');
var ɵngcc3 = require('./bottomnavigation/templates/item-template.directive');
var ɵngcc4 = require('@angular/common');
var templateDirectives = [item_template_directive_1.BottomNavigationItemTemplateDirective];
var exportedComponents = [bottomnavigation_component_1.BottomNavigationComponent].concat(templateDirectives);
var declarations = [bottomnavigation_item_component_1.BottomNavigationItemComponent].concat(exportedComponents);
/**
 * Represents the [NgModule]({{ site.data.urls.angular['ngmoduleapi'] }})
 * definition for the BottomNavigation component.
 * @example
 *
 * ```ts-no-run
 * // Import the BottomNavigation module
 * import { BottomNavigationModule } from '@progress/kendo-angular-navigation';
 *
 * // The browser platform with a compiler
 * import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 * import { BrowserModule } from '@angular/platform-browser';
 *
 * import { NgModule } from '@angular/core';
 *
 * // Import the app component
 * import { AppComponent } from './app.component';
 *
 * // Define the app module
 * _@NgModule({
 *     declarations: [AppComponent], // declare app component
 *     imports:      [BrowserModule, BottomNavigationModule], // import BottomNavigation module
 *     bootstrap:    [AppComponent]
 * })
 * export class AppModule {}
 *
 * // Compile and launch the module
 * platformBrowserDynamic().bootstrapModule(AppModule);
 *
 * ```
 */
var BottomNavigationModule = /** @class */ (function () {
    function BottomNavigationModule() {
    }
BottomNavigationModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BottomNavigationModule });
BottomNavigationModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function BottomNavigationModule_Factory(t) { return new (t || BottomNavigationModule)(); }, imports: [[common_1.CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BottomNavigationModule, { declarations: [ɵngcc1.BottomNavigationItemComponent, ɵngcc2.BottomNavigationComponent, ɵngcc3.BottomNavigationItemTemplateDirective], imports: [ɵngcc4.CommonModule], exports: [ɵngcc2.BottomNavigationComponent, ɵngcc3.BottomNavigationItemTemplateDirective] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BottomNavigationModule, [{
        type: core_1.NgModule,
        args: [{
                declarations: [declarations],
                exports: [exportedComponents],
                imports: [common_1.CommonModule]
            }]
    }], function () { return []; }, null); })();
    return BottomNavigationModule;
}());
exports.BottomNavigationModule = BottomNavigationModule;
