// @progress/kendo-angular-navigation/dist/npm/breadcrumb.module.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var kendo_angular_common_1 = require("@progress/kendo-angular-common");
var breadcrumb_component_1 = require("./breadcrumb/breadcrumb.component");
var breadcrumb_item_component_1 = require("./breadcrumb/breadcrumb-item.component");
var item_template_directive_1 = require("./breadcrumb/template-directives/item-template.directive");
var separator_directive_1 = require("./breadcrumb/template-directives/separator.directive");
var list_component_1 = require("./breadcrumb/list.component");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./breadcrumb/breadcrumb.component');
var ɵngcc2 = require('./breadcrumb/breadcrumb-item.component');
var ɵngcc3 = require('./breadcrumb/list.component');
var ɵngcc4 = require('./breadcrumb/template-directives/item-template.directive');
var ɵngcc5 = require('./breadcrumb/template-directives/separator.directive');
var ɵngcc6 = require('@angular/common');
var ɵngcc7 = require('@progress/kendo-angular-common');
var templateDirectives = [
    item_template_directive_1.BreadCrumbItemTemplateDirective
];
var exportedModules = [
    breadcrumb_component_1.BreadCrumbComponent,
    breadcrumb_item_component_1.BreadCrumbItemComponent,
    list_component_1.BreadCrumbListComponent
].concat(templateDirectives);
var declarations = exportedModules.concat([
    separator_directive_1.BreadCrumbSeparatorDirective
]);
/**
 * Represents the [NgModule]({{ site.data.urls.angular['ngmoduleapi'] }})
 * definition for the BreadCrumb component.
 * @example
 *
 * ```ts-no-run
 * // Import the BreadCrumb module
 * import { BreadCrumbModule } from '@progress/kendo-angular-navigation';
 *
 * // The browser platform with a compiler
 * import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 * import { BrowserModule } from '@angular/platform-browser';
 *
 * import { NgModule } from '@angular/core';
 *
 * // Import the app component
 * import { AppComponent } from './app.component';
 *
 * // Define the app module
 * _@NgModule({
 *     declarations: [AppComponent], // declare app component
 *     imports:      [BrowserModule, BreadCrumbModule], // import BreadCrumb module
 *     bootstrap:    [AppComponent]
 * })
 * export class AppModule {}
 *
 * // Compile and launch the module
 * platformBrowserDynamic().bootstrapModule(AppModule);
 *
 * ```
 */
var BreadCrumbModule = /** @class */ (function () {
    function BreadCrumbModule() {
    }
BreadCrumbModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BreadCrumbModule });
BreadCrumbModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function BreadCrumbModule_Factory(t) { return new (t || BreadCrumbModule)(); }, imports: [[common_1.CommonModule, kendo_angular_common_1.ResizeSensorModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BreadCrumbModule, { declarations: [ɵngcc1.BreadCrumbComponent, ɵngcc2.BreadCrumbItemComponent, ɵngcc3.BreadCrumbListComponent, ɵngcc4.BreadCrumbItemTemplateDirective, ɵngcc5.BreadCrumbSeparatorDirective], imports: [ɵngcc6.CommonModule, ɵngcc7.ResizeSensorModule], exports: [ɵngcc1.BreadCrumbComponent, ɵngcc2.BreadCrumbItemComponent, ɵngcc3.BreadCrumbListComponent, ɵngcc4.BreadCrumbItemTemplateDirective] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BreadCrumbModule, [{
        type: core_1.NgModule,
        args: [{
                declarations: [declarations],
                exports: [exportedModules],
                imports: [common_1.CommonModule, kendo_angular_common_1.ResizeSensorModule]
            }]
    }], function () { return []; }, null); })();
    return BreadCrumbModule;
}());
exports.BreadCrumbModule = BreadCrumbModule;
