// @progress/kendo-angular-navigation/dist/npm/navigation.module.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@angular/core");
var appbar_module_1 = require("./appbar.module");
var breadcrumb_module_1 = require("./breadcrumb.module");
var bottomnavigation_module_1 = require("./bottomnavigation.module");
/**
 * Represents the [NgModule]({{ site.data.urls.angular['ngmoduleapi'] }})
 * definition for the Navigation components.
 *
 * @example
 *
 * ```ts-no-run
 * // Import the Navigation module
 * import { NavigationModule } from '@progress/kendo-angular-navigation';
 *
 * // The browser platform with a compiler
 * import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 *
 * import { NgModule } from '@angular/core';
 *
 * // Import the app component
 * import { AppComponent } from './app.component';
 *
 * // Define the app module
 * _@NgModule({
 *     declarations: [AppComponent], // declare app component
 *     imports:      [BrowserModule, NavigationModule], // import Navigation module
 *     bootstrap:    [AppComponent]
 * })
 * export class AppModule {}
 *
 * // Compile and launch the module
 * platformBrowserDynamic().bootstrapModule(AppModule);
 *
 * ```
 */
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./appbar.module');
var ɵngcc2 = require('./breadcrumb.module');
var ɵngcc3 = require('./bottomnavigation.module');
var NavigationModule = /** @class */ (function () {
    function NavigationModule() {
    }
NavigationModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NavigationModule });
NavigationModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NavigationModule_Factory(t) { return new (t || NavigationModule)(); }, imports: [ɵngcc1.AppBarModule, ɵngcc2.BreadCrumbModule, ɵngcc3.BottomNavigationModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NavigationModule, { exports: [ɵngcc1.AppBarModule, ɵngcc2.BreadCrumbModule, ɵngcc3.BottomNavigationModule] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NavigationModule, [{
        type: core_1.NgModule,
        args: [{
                exports: [
                    appbar_module_1.AppBarModule,
                    breadcrumb_module_1.BreadCrumbModule,
                    bottomnavigation_module_1.BottomNavigationModule
                ]
            }]
    }], function () { return []; }, null); })();
    return NavigationModule;
}());
exports.NavigationModule = NavigationModule;
