// @progress/kendo-angular-navigation/dist/npm/bottomnavigation/templates/item-template.directive.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@angular/core");
/**
 * Represents a template that defines the item content of the BottomNavigation.
 * To define the template, nest an `<ng-template>` tag
 * with the `kendoBottomNavigationItemTemplate` directive inside the `<kendo-bottomnavigation>` tag ([see example]({% slug templates_bottomnavigation %})).
 */
var ɵngcc0 = require('@angular/core');
var BottomNavigationItemTemplateDirective = /** @class */ (function () {
    function BottomNavigationItemTemplateDirective(templateRef) {
        this.templateRef = templateRef;
    }
    BottomNavigationItemTemplateDirective = tslib_1.__decorate([ tslib_1.__param(0, core_1.Optional()),
        tslib_1.__metadata("design:paramtypes", [core_1.TemplateRef])
    ], BottomNavigationItemTemplateDirective);
BottomNavigationItemTemplateDirective.ɵfac = function BottomNavigationItemTemplateDirective_Factory(t) { return new (t || BottomNavigationItemTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef, 8)); };
BottomNavigationItemTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BottomNavigationItemTemplateDirective, selectors: [["", "kendoBottomNavigationItemTemplate", ""]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BottomNavigationItemTemplateDirective, [{
        type: core_1.Directive,
        args: [{
                selector: '[kendoBottomNavigationItemTemplate]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef, decorators: [{
                type: core_1.Optional
            }] }]; }, null); })();
    return BottomNavigationItemTemplateDirective;
}());
exports.BottomNavigationItemTemplateDirective = BottomNavigationItemTemplateDirective;
