// @progress/kendo-angular-navigation/dist/es/bottomnavigation.module.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
import * as tslib_1 from "tslib";
import { BottomNavigationItemComponent } from './bottomnavigation/bottomnavigation-item.component';
import { BottomNavigationItemTemplateDirective } from './bottomnavigation/templates/item-template.directive';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BottomNavigationComponent } from './bottomnavigation/bottomnavigation.component';
import * as ɵngcc0 from '@angular/core';
var templateDirectives = [BottomNavigationItemTemplateDirective];
var exportedComponents = [BottomNavigationComponent].concat(templateDirectives);
var declarations = [BottomNavigationItemComponent].concat(exportedComponents);
/**
 * Represents the [NgModule]({{ site.data.urls.angular['ngmoduleapi'] }})
 * definition for the BottomNavigation component.
 * @example
 *
 * ```ts-no-run
 * // Import the BottomNavigation module
 * import { BottomNavigationModule } from '@progress/kendo-angular-navigation';
 *
 * // The browser platform with a compiler
 * import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
 * import { BrowserModule } from '@angular/platform-browser';
 *
 * import { NgModule } from '@angular/core';
 *
 * // Import the app component
 * import { AppComponent } from './app.component';
 *
 * // Define the app module
 * _@NgModule({
 *     declarations: [AppComponent], // declare app component
 *     imports:      [BrowserModule, BottomNavigationModule], // import BottomNavigation module
 *     bootstrap:    [AppComponent]
 * })
 * export class AppModule {}
 *
 * // Compile and launch the module
 * platformBrowserDynamic().bootstrapModule(AppModule);
 *
 * ```
 */
var BottomNavigationModule = /** @class */ (function () {
    function BottomNavigationModule() {
    }
BottomNavigationModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BottomNavigationModule });
BottomNavigationModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function BottomNavigationModule_Factory(t) { return new (t || BottomNavigationModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BottomNavigationModule, { declarations: function () { return [BottomNavigationItemComponent, BottomNavigationComponent, BottomNavigationItemTemplateDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [BottomNavigationComponent, BottomNavigationItemTemplateDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BottomNavigationModule, [{
        type: NgModule,
        args: [{
                declarations: [declarations],
                exports: [exportedComponents],
                imports: [CommonModule]
            }]
    }], function () { return []; }, null); })();
    return BottomNavigationModule;
}());
export { BottomNavigationModule };
