// mat-select-filter/bundles/mat-select-filter.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/forms'),require('@angular/common'),require('@angular/material/progress-spinner'),exports, require('@angular/core'), require('@angular/forms'), require('@angular/cdk/keycodes'), require('@angular/material/progress-spinner'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('mat-select-filter', ['@angular/core','@angular/forms','@angular/common','@angular/material/progress-spinner','exports', '@angular/core', '@angular/forms', '@angular/cdk/keycodes', '@angular/material/progress-spinner', '@angular/common'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global.ng.core,global.ng.forms,global.ng.common,global.ng.material.progressSpinner,global['mat-select-filter'] = {}, global.ng.core, global.ng.forms, global.ng.cdk.keycodes, global.ng.material.progressSpinner, global.ng.common));
}(this, (function (ɵngcc0,ɵngcc1,ɵngcc2,ɵngcc3,exports, i0, forms, keycodes, progressSpinner, common) { 
var _c0 = ["input"];
function MatSelectFilterComponent_mat_spinner_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "mat-spinner", 5);
} }
function MatSelectFilterComponent_div_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r2.noResultsMessage, "\n");
} }
var _c1 = function (a0) { return { "background-color": a0 }; };
'use strict';

    var MatSelectFilterService = /** @class */ (function () {
        function MatSelectFilterService() {
        }
MatSelectFilterService.ɵfac = function MatSelectFilterService_Factory(t) { return new (t || MatSelectFilterService)(); };
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(MatSelectFilterService, [{
        type: i0.Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();
        return MatSelectFilterService;
    }());
    MatSelectFilterService.ɵprov = i0.ɵɵdefineInjectable({ factory: function MatSelectFilterService_Factory() { return new MatSelectFilterService(); }, token: MatSelectFilterService, providedIn: "root" });
    MatSelectFilterService.ctorParameters = function () { return []; };

    var MatSelectFilterComponent = /** @class */ (function () {
        function MatSelectFilterComponent(fb) {
            this.showSpinner = true;
            this.noResultsMessage = 'No results';
            this.noResults = false;
            this.localSpinner = false;
            this.filteredReturn = new i0.EventEmitter();
            this.filteredItems = [];
            this.searchForm = fb.group({
                value: ''
            });
        }
        MatSelectFilterComponent.prototype.ngOnInit = function () {
            var _this = this;
            this.searchForm.valueChanges.subscribe(function (value) {
                if (_this.showSpinner) {
                    _this.localSpinner = true;
                }
                if (value['value']) {
                    // IF THE DISPLAY MEMBER INPUT IS SET WE CHECK THE SPECIFIC PROPERTY
                    if (_this.displayMember == null) {
                        _this.filteredItems = _this.array.filter(function (name) { return name.toLowerCase().includes(value['value'].toLowerCase()); });
                        // OTHERWISE, WE CHECK THE ENTIRE STRING
                    }
                    else if (_this.hasGroup && _this.groupArrayName && _this.displayMember) {
                        _this.filteredItems = _this.array.map(function (a) {
                            var objCopy = Object.assign({}, a);
                            objCopy[_this.groupArrayName] = objCopy[_this.groupArrayName].filter(function (g) { return g[_this.displayMember].toLowerCase().includes(value['value'].toLowerCase()); });
                            return objCopy;
                        }).filter(function (x) { return x[_this.groupArrayName].length > 0; });
                    }
                    else {
                        _this.filteredItems = _this.array.filter(function (name) { return name[_this.displayMember].toLowerCase().includes(value['value'].toLowerCase()); });
                    }
                    // NO RESULTS VALIDATION
                    _this.noResults = _this.filteredItems == null || _this.filteredItems.length === 0;
                }
                else {
                    _this.filteredItems = _this.array.slice();
                    _this.noResults = false;
                }
                _this.filteredReturn.emit(_this.filteredItems);
                setTimeout(function () {
                    if (_this.showSpinner) {
                        _this.localSpinner = false;
                    }
                }, 2000);
            });
            setTimeout(function () {
                _this.input.nativeElement.focus();
            }, 500);
            if (!this.placeholder) {
                this.placeholder = 'Search...';
            }
        };
        MatSelectFilterComponent.prototype.handleKeydown = function (event) {
            // PREVENT PROPAGATION FOR ALL ALPHANUMERIC CHARACTERS IN ORDER TO AVOID SELECTION ISSUES
            if ((event.key && event.key.length === 1) ||
                (event.keyCode >= keycodes.A && event.keyCode <= keycodes.Z) ||
                (event.keyCode >= keycodes.ZERO && event.keyCode <= keycodes.NINE) ||
                (event.keyCode === keycodes.SPACE)) {
                event.stopPropagation();
            }
        };
        MatSelectFilterComponent.prototype.ngOnDestroy = function () {
            this.filteredReturn.emit(this.array);
        };
MatSelectFilterComponent.ɵfac = function MatSelectFilterComponent_Factory(t) { return new (t || MatSelectFilterComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormBuilder)); };
MatSelectFilterComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MatSelectFilterComponent, selectors: [["mat-select-filter"]], viewQuery: function MatSelectFilterComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, 3);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.input = _t.first);
    } }, inputs: { showSpinner: "showSpinner", noResultsMessage: "noResultsMessage", placeholder: "placeholder", array: "array", color: "color", displayMember: "displayMember", hasGroup: "hasGroup", groupArrayName: "groupArrayName" }, outputs: { filteredReturn: "filteredReturn" }, decls: 6, vars: 7, consts: [[1, "mat-filter", 3, "formGroup", "ngStyle"], ["matInput", "", "formControlName", "value", 1, "mat-filter-input", 3, "placeholder", "keydown"], ["input", ""], ["class", "spinner", "diameter", "16", 4, "ngIf"], ["class", "noResultsMessage", 4, "ngIf"], ["diameter", "16", 1, "spinner"], [1, "noResultsMessage"]], template: function MatSelectFilterComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "form", 0);
        ɵngcc0.ɵɵelementStart(1, "div");
        ɵngcc0.ɵɵelementStart(2, "input", 1, 2);
        ɵngcc0.ɵɵlistener("keydown", function MatSelectFilterComponent_Template_input_keydown_2_listener($event) { return ctx.handleKeydown($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(4, MatSelectFilterComponent_mat_spinner_4_Template, 1, 0, "mat-spinner", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(5, MatSelectFilterComponent_div_5_Template, 2, 1, "div", 4);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("formGroup", ctx.searchForm)("ngStyle", ɵngcc0.ɵɵpureFunction1(5, _c1, ctx.color ? ctx.color : "white"));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵpropertyInterpolate("placeholder", ctx.placeholder);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.localSpinner);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.noResults);
    } }, directives: [ɵngcc1.ɵangular_packages_forms_forms_y, ɵngcc1.NgControlStatusGroup, ɵngcc1.FormGroupDirective, ɵngcc2.NgStyle, ɵngcc1.DefaultValueAccessor, ɵngcc1.NgControlStatus, ɵngcc1.FormControlName, ɵngcc2.NgIf, ɵngcc3.MatSpinner], styles: [".mat-filter[_ngcontent-%COMP%]{position:sticky;top:0;border-bottom:1px solid grey;z-index:100;font-size:inherit;box-shadow:none;border-radius:0;padding:16px;box-sizing:border-box}.mat-filter-input[_ngcontent-%COMP%]{-webkit-appearance:none;-moz-appearance:none;appearance:none;outline:none;border:0;background-color:unset;color:grey;width:100%}.spinner[_ngcontent-%COMP%]{position:absolute;right:16px;top:calc(50% - 8px)}.noResultsMessage[_ngcontent-%COMP%]{margin-top:10px;font-family:Roboto,Helvetica Neue,sans-serif;font-size:16px}"] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(MatSelectFilterComponent, [{
        type: i0.Component,
        args: [{
                selector: 'mat-select-filter',
                template: "\n  <form [formGroup]=\"searchForm\" class=\"mat-filter\" [ngStyle]=\"{'background-color': color ? color : 'white'}\">\n  <div>\n  <input #input class=\"mat-filter-input\" matInput placeholder=\"{{placeholder}}\" formControlName=\"value\" (keydown)=\"handleKeydown($event)\">\n    <mat-spinner *ngIf=\"localSpinner\" class=\"spinner\" diameter=\"16\"></mat-spinner>\n  </div>\n  <div *ngIf=\"noResults\"\n     class=\"noResultsMessage\">\n  {{noResultsMessage}}\n</div>\n</form>\n  ",
                styles: [".mat-filter{position:sticky;top:0;border-bottom:1px solid grey;z-index:100;font-size:inherit;box-shadow:none;border-radius:0;padding:16px;box-sizing:border-box}.mat-filter-input{-webkit-appearance:none;-moz-appearance:none;appearance:none;outline:none;border:0;background-color:unset;color:grey;width:100%}.spinner{position:absolute;right:16px;top:calc(50% - 8px)}.noResultsMessage{margin-top:10px;font-family:Roboto,Helvetica Neue,sans-serif;font-size:16px}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormBuilder }]; }, { showSpinner: [{
            type: i0.Input,
            args: ['showSpinner']
        }], noResultsMessage: [{
            type: i0.Input,
            args: ['noResultsMessage']
        }], filteredReturn: [{
            type: i0.Output
        }], placeholder: [{
            type: i0.Input,
            args: ['placeholder']
        }], input: [{
            type: i0.ViewChild,
            args: ['input', { static: true }]
        }], array: [{
            type: i0.Input,
            args: ['array']
        }], color: [{
            type: i0.Input,
            args: ['color']
        }], displayMember: [{
            type: i0.Input,
            args: ['displayMember']
        }], hasGroup: [{
            type: i0.Input,
            args: ['hasGroup']
        }], groupArrayName: [{
            type: i0.Input,
            args: ['groupArrayName']
        }] }); })();
        return MatSelectFilterComponent;
    }());
    MatSelectFilterComponent.ctorParameters = function () { return [
        { type: forms.FormBuilder }
    ]; };
    MatSelectFilterComponent.propDecorators = {
        input: [{ type: i0.ViewChild, args: ['input', { static: true },] }],
        array: [{ type: i0.Input, args: ['array',] }],
        placeholder: [{ type: i0.Input, args: ['placeholder',] }],
        color: [{ type: i0.Input, args: ['color',] }],
        displayMember: [{ type: i0.Input, args: ['displayMember',] }],
        showSpinner: [{ type: i0.Input, args: ['showSpinner',] }],
        noResultsMessage: [{ type: i0.Input, args: ['noResultsMessage',] }],
        hasGroup: [{ type: i0.Input, args: ['hasGroup',] }],
        groupArrayName: [{ type: i0.Input, args: ['groupArrayName',] }],
        filteredReturn: [{ type: i0.Output }]
    };

    var MatSelectFilterModule = /** @class */ (function () {
        function MatSelectFilterModule() {
        }
MatSelectFilterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MatSelectFilterModule });
MatSelectFilterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MatSelectFilterModule_Factory(t) { return new (t || MatSelectFilterModule)(); }, imports: [[
            common.CommonModule,
            forms.FormsModule,
            forms.ReactiveFormsModule,
            progressSpinner.MatProgressSpinnerModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MatSelectFilterModule, { declarations: [MatSelectFilterComponent], imports: [ɵngcc2.CommonModule, ɵngcc1.FormsModule, ɵngcc1.ReactiveFormsModule, ɵngcc3.MatProgressSpinnerModule], exports: [MatSelectFilterComponent] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(MatSelectFilterModule, [{
        type: i0.NgModule,
        args: [{
                declarations: [MatSelectFilterComponent],
                imports: [
                    common.CommonModule,
                    forms.FormsModule,
                    forms.ReactiveFormsModule,
                    progressSpinner.MatProgressSpinnerModule
                ],
                exports: [MatSelectFilterComponent]
            }]
    }], function () { return []; }, null); })();
        return MatSelectFilterModule;
    }());

    /*
     * Public API Surface of mat-select-filter
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.MatSelectFilterComponent = MatSelectFilterComponent;
    exports.MatSelectFilterModule = MatSelectFilterModule;
    exports.MatSelectFilterService = MatSelectFilterService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
