// @progress/kendo-angular-navigation/dist/es/breadcrumb/template-directives/item-template.directive.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
import * as tslib_1 from "tslib";
import { Directive, Optional, TemplateRef } from '@angular/core';
/**
 * Represents a template that defines the content of a Breadcrumb item.
 * To define the template, nest an `<ng-template>` tag with the `kendoBreadCrumbItemTemplate` directive inside the `<kendo-breadcrumb>` tag.
 *
 * For more information and example refer to the [Templates]({% slug templates_breadcrumb %}) article.
 */
import * as ɵngcc0 from '@angular/core';
var BreadCrumbItemTemplateDirective = /** @class */ (function () {
    function BreadCrumbItemTemplateDirective(templateRef) {
        this.templateRef = templateRef;
    }
    BreadCrumbItemTemplateDirective = tslib_1.__decorate([ tslib_1.__param(0, Optional()),
        tslib_1.__metadata("design:paramtypes", [TemplateRef])
    ], BreadCrumbItemTemplateDirective);
BreadCrumbItemTemplateDirective.ɵfac = function BreadCrumbItemTemplateDirective_Factory(t) { return new (t || BreadCrumbItemTemplateDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef, 8)); };
BreadCrumbItemTemplateDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BreadCrumbItemTemplateDirective, selectors: [["", "kendoBreadCrumbItemTemplate", ""]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BreadCrumbItemTemplateDirective, [{
        type: Directive,
        args: [{
                selector: '[kendoBreadCrumbItemTemplate]'
            }]
    }], function () { return [{ type: ɵngcc0.TemplateRef, decorators: [{
                type: Optional
            }] }]; }, null); })();
    return BreadCrumbItemTemplateDirective;
}());
export { BreadCrumbItemTemplateDirective };
