// @progress/kendo-angular-navigation/dist/es/breadcrumb/util.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
var getCollapsed = function (itemComponent) { return itemComponent.item.context.collapsed; };
var ɵ0 = getCollapsed;
/**
 * @hidden
 */
export var collapsed = function (itemComponent) { return getCollapsed(itemComponent) === true; };
/**
 * @hidden
 */
export var expanded = function (itemComponent) { return getCollapsed(itemComponent) === false; };
var toggleFirst = function (collapsed) { return function (itemComponents) {
    return (itemComponents.find(function (ic) { return getCollapsed(ic) === collapsed; }).item.context.collapsed = !collapsed);
}; };
var ɵ1 = toggleFirst;
/**
 * @hidden
 */
export var collapseFirst = toggleFirst(false);
/**
 * @hidden
 */
export var expandFirst = toggleFirst(true);
export { ɵ0, ɵ1 };
