// @progress/kendo-angular-navigation/dist/npm/appbar/appbar-spacer.component.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@angular/core");
var util_1 = require("../common/util");
/**
 * Represents the [Kendo UI AppBarSpacer component for Angular]({% slug contentarrangement_appbar %}#toc-spacings).
 * Used to give additional white space between the AppBar sections and provides a way for customizing its width.
 *
 * @example
 *
 * ```ts-no-run
 *  * _@Component({
 *    selector: 'my-app',
 *    template: `
 *        <kendo-appbar>
 *            <kendo-appbar-section>
 *                 <button class="k-button k-button-clear">
 *                     <kendo-icon [name]="'menu'"></kendo-icon>
 *                 </button>
 *            </kendo-appbar-section>
 *
 *            <kendo-appbar-spacer></kendo-appbar-spacer>
 *
 *            <kendo-appbar-section>
 *                <h2>Page Title</h2>
 *            </kendo-appbar-section>
 *        </kendo-appbar>
 *    `
 * })
 * class AppComponent {}
 * ```
 */
var ɵngcc0 = require('@angular/core');
var AppBarSpacerComponent = /** @class */ (function () {
    function AppBarSpacerComponent(renderer, element) {
        this.renderer = renderer;
        this.element = element;
        this.hostClass = true;
    }
    Object.defineProperty(AppBarSpacerComponent.prototype, "sizedClass", {
        get: function () {
            return util_1.isPresent(this.width);
        },
        enumerable: true,
        configurable: true
    });
    AppBarSpacerComponent.prototype.ngAfterViewInit = function () {
        if (util_1.isPresent(this.width)) {
            var element = this.element.nativeElement;
            this.renderer.setStyle(element, 'flexBasis', this.width);
        }
    };
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar-spacer'),
        tslib_1.__metadata("design:type", Boolean)
    ], AppBarSpacerComponent.prototype, "hostClass", void 0);
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar-spacer-sized'),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [])
    ], AppBarSpacerComponent.prototype, "sizedClass", null);
    tslib_1.__decorate([
        core_1.Input(),
        tslib_1.__metadata("design:type", String)
    ], AppBarSpacerComponent.prototype, "width", void 0);
    AppBarSpacerComponent = tslib_1.__decorate([ tslib_1.__metadata("design:paramtypes", [core_1.Renderer2, core_1.ElementRef])
    ], AppBarSpacerComponent);
AppBarSpacerComponent.ɵfac = function AppBarSpacerComponent_Factory(t) { return new (t || AppBarSpacerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
AppBarSpacerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AppBarSpacerComponent, selectors: [["kendo-appbar-spacer"]], hostVars: 4, hostBindings: function AppBarSpacerComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("k-appbar-spacer", ctx.hostClass)("k-appbar-spacer-sized", ctx.sizedClass);
    } }, inputs: { width: "width" }, decls: 0, vars: 0, template: function AppBarSpacerComponent_Template(rf, ctx) { }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AppBarSpacerComponent, [{
        type: core_1.Component,
        args: [{
                selector: 'kendo-appbar-spacer',
                template: ""
            }]
    }], function () { return [{ type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ElementRef }]; }, { hostClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar-spacer']
        }], sizedClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar-spacer-sized']
        }], width: [{
            type: core_1.Input
        }] }); })();
    return AppBarSpacerComponent;
}());
exports.AppBarSpacerComponent = AppBarSpacerComponent;
