// @progress/kendo-angular-navigation/dist/npm/appbar/appbar.component.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@angular/core");
var kendo_angular_l10n_1 = require("@progress/kendo-angular-l10n");
var kendo_licensing_1 = require("@progress/kendo-licensing");
var package_metadata_1 = require("../package-metadata");
/**
 * Represents the [Kendo UI AppBar component for Angular]({% slug overview_appbar %}).
 * Used to display information, actions, branding titles and additional navigation on the current screen.
 */
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('@progress/kendo-angular-l10n');

var _c0 = ["*"];
var AppBarComponent = /** @class */ (function () {
    function AppBarComponent(localizationService) {
        var _this = this;
        this.localizationService = localizationService;
        this.hostClass = true;
        /**
         * Specifies the position of the AppBar
         * ([see example]({% slug positioning_appbar %}#toc-position)).
         *
         * * The possible values are:
         * * `top` (Default)&mdash;Positions the AppBar at the top of the content.
         *    Setting the `position` property to `top` requires adding the Appbar component before the page content.
         *    The position property applies CSS `top: 0` style in [`fixed mode`]({% slug api_navigation_appbarcomponent %}#toc-positionMode) and also adds a `box-shadow` to the bottom of the AppBar.
         *
         * * `bottom`&mdash;Positions the AppBar at the bottom of the content.
         *    Setting the `position` property to `bottom` requires adding the Appbar component after the page content.
         *   The position property applies CSS `bottom: 0` style in [`fixed mode`]({% slug api_navigation_appbarcomponent %}#toc-positionMode) and also adds a `box-shadow ` to the top of the AppBar.
         *
         */
        this.position = 'top';
        /**
         * Specifies the positionMode of the AppBar
         * ([see example]({% slug positioning_appbar %}#toc-positionmode)).
         *
         * * The possible values are:
         * * `static` (Default)&mdash;Does not position the AppBar in any special way. It is positioned according to the normal flow of the page.
         * * `sticky`&mdash;Positions the AppBar based on the user's scroll position. A sticky element toggles between static and fixed CSS [`position`](https://developer.mozilla.org/en-US/docs/Web/CSS/position) property, depending on the scroll position.
         * * `fixed`&mdash;Positions the AppBar relative to the viewport. It always stays in the same place even if the page is scrolled.
         */
        this.positionMode = 'static';
        /**
         * Specifies the theme color of the AppBar.
         * The theme color will be applied as background color of the component.
         *
         *
         * * The possible values are:
         * * `light` (Default)&mdash;Applies coloring based on light theme color.
         * * `dark`&mdash;Applies coloring based on dark theme color.
         * * `inherit`&mdash; Applies inherited coloring value.
         *
         */
        this.themeColor = 'light';
        this.rtl = false;
        kendo_licensing_1.validatePackage(package_metadata_1.packageMetadata);
        this.dynamicRTLSubscription = this.localizationService.changes.subscribe(function (_a) {
            var rtl = _a.rtl;
            _this.rtl = rtl;
            _this.direction = _this.rtl ? 'rtl' : 'ltr';
        });
    }
    Object.defineProperty(AppBarComponent.prototype, "topClass", {
        get: function () {
            return this.position === 'top';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AppBarComponent.prototype, "bottomClass", {
        get: function () {
            return this.position === 'bottom';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AppBarComponent.prototype, "stickyClass", {
        get: function () {
            return this.positionMode === 'sticky';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AppBarComponent.prototype, "fixedClass", {
        get: function () {
            return this.positionMode === 'fixed';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AppBarComponent.prototype, "staticClass", {
        get: function () {
            return this.positionMode === 'static';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AppBarComponent.prototype, "lightClass", {
        get: function () {
            return this.themeColor === 'light';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AppBarComponent.prototype, "darkClass", {
        get: function () {
            return this.themeColor === 'dark';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AppBarComponent.prototype, "inheritClass", {
        get: function () {
            return this.themeColor === 'inherit';
        },
        enumerable: true,
        configurable: true
    });
    AppBarComponent.prototype.ngOnDestroy = function () {
        if (this.dynamicRTLSubscription) {
            this.dynamicRTLSubscription.unsubscribe();
        }
    };
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar'),
        tslib_1.__metadata("design:type", Boolean)
    ], AppBarComponent.prototype, "hostClass", void 0);
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar-top'),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [])
    ], AppBarComponent.prototype, "topClass", null);
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar-bottom'),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [])
    ], AppBarComponent.prototype, "bottomClass", null);
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar-sticky'),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [])
    ], AppBarComponent.prototype, "stickyClass", null);
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar-fixed'),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [])
    ], AppBarComponent.prototype, "fixedClass", null);
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar-static'),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [])
    ], AppBarComponent.prototype, "staticClass", null);
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar-light'),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [])
    ], AppBarComponent.prototype, "lightClass", null);
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar-dark'),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [])
    ], AppBarComponent.prototype, "darkClass", null);
    tslib_1.__decorate([
        core_1.HostBinding('class.k-appbar-inherit'),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [])
    ], AppBarComponent.prototype, "inheritClass", null);
    tslib_1.__decorate([
        core_1.HostBinding('attr.dir'),
        tslib_1.__metadata("design:type", String)
    ], AppBarComponent.prototype, "direction", void 0);
    tslib_1.__decorate([
        core_1.Input(),
        tslib_1.__metadata("design:type", String)
    ], AppBarComponent.prototype, "position", void 0);
    tslib_1.__decorate([
        core_1.Input(),
        tslib_1.__metadata("design:type", String)
    ], AppBarComponent.prototype, "positionMode", void 0);
    tslib_1.__decorate([
        core_1.Input(),
        tslib_1.__metadata("design:type", String)
    ], AppBarComponent.prototype, "themeColor", void 0);
    AppBarComponent = tslib_1.__decorate([ tslib_1.__metadata("design:paramtypes", [kendo_angular_l10n_1.LocalizationService])
    ], AppBarComponent);
AppBarComponent.ɵfac = function AppBarComponent_Factory(t) { return new (t || AppBarComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.LocalizationService)); };
AppBarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AppBarComponent, selectors: [["kendo-appbar"]], hostVars: 19, hostBindings: function AppBarComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("dir", ctx.direction);
        ɵngcc0.ɵɵclassProp("k-appbar", ctx.hostClass)("k-appbar-top", ctx.topClass)("k-appbar-bottom", ctx.bottomClass)("k-appbar-sticky", ctx.stickyClass)("k-appbar-fixed", ctx.fixedClass)("k-appbar-static", ctx.staticClass)("k-appbar-light", ctx.lightClass)("k-appbar-dark", ctx.darkClass)("k-appbar-inherit", ctx.inheritClass);
    } }, inputs: { position: "position", positionMode: "positionMode", themeColor: "themeColor" }, exportAs: ["kendoAppBar"], features: [ɵngcc0.ɵɵProvidersFeature([
            kendo_angular_l10n_1.LocalizationService,
            {
                provide: kendo_angular_l10n_1.L10N_PREFIX,
                useValue: 'kendo.appbar.component'
            }
        ])], ngContentSelectors: _c0, decls: 1, vars: 0, template: function AppBarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AppBarComponent, [{
        type: core_1.Component,
        args: [{
                exportAs: 'kendoAppBar',
                selector: 'kendo-appbar',
                template: "\n        <ng-content></ng-content>\n    ",
                providers: [
                    kendo_angular_l10n_1.LocalizationService,
                    {
                        provide: kendo_angular_l10n_1.L10N_PREFIX,
                        useValue: 'kendo.appbar.component'
                    }
                ]
            }]
    }], function () { return [{ type: ɵngcc1.LocalizationService }]; }, { hostClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar']
        }], position: [{
            type: core_1.Input
        }], positionMode: [{
            type: core_1.Input
        }], themeColor: [{
            type: core_1.Input
        }], topClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar-top']
        }], bottomClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar-bottom']
        }], stickyClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar-sticky']
        }], fixedClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar-fixed']
        }], staticClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar-static']
        }], lightClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar-light']
        }], darkClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar-dark']
        }], inheritClass: [{
            type: core_1.HostBinding,
            args: ['class.k-appbar-inherit']
        }], direction: [{
            type: core_1.HostBinding,
            args: ['attr.dir']
        }] }); })();
    return AppBarComponent;
}());
exports.AppBarComponent = AppBarComponent;
