// @angular-material-components/file-input/esm2015/lib/file-input.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatIconModule } from '@angular/material/icon';
import { MatInputModule } from '@angular/material/input';
import { NgxMatFileInputComponent, NgxMatFileInputIcon } from './file-input.component';
import * as ɵngcc0 from '@angular/core';
export class NgxMatFileInputModule {
}
NgxMatFileInputModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxMatFileInputModule });
NgxMatFileInputModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxMatFileInputModule_Factory(t) { return new (t || NgxMatFileInputModule)(); }, imports: [[
            CommonModule,
            MatButtonModule,
            MatIconModule,
            MatFormFieldModule,
            MatInputModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxMatFileInputModule, { declarations: function () { return [NgxMatFileInputComponent,
        NgxMatFileInputIcon]; }, imports: function () { return [CommonModule,
        MatButtonModule,
        MatIconModule,
        MatFormFieldModule,
        MatInputModule]; }, exports: function () { return [NgxMatFileInputComponent,
        NgxMatFileInputIcon]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxMatFileInputModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    NgxMatFileInputComponent,
                    NgxMatFileInputIcon
                ],
                imports: [
                    CommonModule,
                    MatButtonModule,
                    MatIconModule,
                    MatFormFieldModule,
                    MatInputModule
                ],
                exports: [
                    NgxMatFileInputComponent,
                    NgxMatFileInputIcon
                ]
            }]
    }], null, null); })();
