// ng-sidebar/__ivy_ngcc__/lib_esmodule/close.directive.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './sidebar.component';
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Directive } from '@angular/core';
import { Sidebar } from './sidebar.component';
var CloseSidebar = (function () {
    function CloseSidebar(_sidebar) {
        this._sidebar = _sidebar;
    }
    CloseSidebar.prototype._onClick = function () {
        if (this._sidebar) {
            this._sidebar.close();
        }
    };
    CloseSidebar = __decorate([ __metadata("design:paramtypes", [Sidebar])
    ], CloseSidebar);
CloseSidebar.ɵfac = function CloseSidebar_Factory(t) { return new (t || CloseSidebar)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Sidebar)); };
CloseSidebar.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CloseSidebar, selectors: [["", "closeSidebar", ""]], hostBindings: function CloseSidebar_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function CloseSidebar_click_HostBindingHandler() { return ctx._onClick(); });
    } } });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CloseSidebar, [{
        type: Directive,
        args: [{
                selector: '[closeSidebar]',
                host: {
                    '(click)': '_onClick()'
                }
            }]
    }], function () { return [{ type: ɵngcc1.Sidebar }]; }, null); })();
    return CloseSidebar;
}());
export { CloseSidebar };
