// @progress/kendo-angular-common/dist/es/events/events.module.js
/**-----------------------------------------------------------------------------------------
* Copyright © 2020 Progress Software Corporation. All rights reserved.
* Licensed under commercial license. See LICENSE.md in the project root for more information
*-------------------------------------------------------------------------------------------*/
import * as tslib_1 from "tslib";
import { NgModule } from '@angular/core';
import { EventsOutsideAngularDirective } from './events-outside-angular.directive';
/**
 * @hidden
 */
import * as ɵngcc0 from '@angular/core';
var EventsModule = /** @class */ (function () {
    function EventsModule() {
    }
EventsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: EventsModule });
EventsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function EventsModule_Factory(t) { return new (t || EventsModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(EventsModule, { declarations: function () { return [EventsOutsideAngularDirective]; }, exports: function () { return [EventsOutsideAngularDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(EventsModule, [{
        type: NgModule,
        args: [{
                declarations: [EventsOutsideAngularDirective],
                exports: [EventsOutsideAngularDirective]
            }]
    }], function () { return []; }, null); })();
    return EventsModule;
}());
export { EventsModule };
